src/js/AppStoreMenu.js

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import Modal from 'react-modal'
+import { connect } from 'react-redux';
+import store from './store';
+import { addAppPendingSetAppProperties } from './actions'
+
+import AppHeader from './AppHeader';
+import StaticIcon from './Templates/Shared/StaticIcon';
+import ConfirmAlert from './ConfirmAlert';
+
+import BCController from './Controllers/BCController';
+import fileSystemController from './Controllers/FileSystemController';
+
+class AppStoreMenuItem extends React.Component {
+    constructor() {
+        super();
+    }
+
+    render() {
+        return (
+            <div className="appstore-menu-item th-b-color th-bb-color-secondary" key={this.props.appName} >
+                <div className="appstore-menu-item__image" >
+                    <img src={this.props.appIcon} />
+                </div>
+                <h2 className="t-large t-medium">{this.props.appName}</h2>
+                <h4 className="t-large t-light">{this.props.appSize}</h4>
+                <div className="svg-wrap" onClick={this.props.onSelect}>
+                    <StaticIcon class="static-icon" image="0x0F" />
+                </div>
+            </div>
+        )
+    }
+}
+
+class AppStoreMenu extends React.Component {
+    constructor() {
+        super();
+
+        this.state = {
+            confirmID: null,
+            appName: null,
+            appIcon: null,
+            appDescription: null
+        }
+
+        this.cancel = this.cancel.bind(this);
+        this.confirm = this.confirm.bind(this);
+    }
+
+    onSelection(appID, name, icon, description) {
+        this.setState((state, props) => {
+            return {
+                confirmID: appID,
+                appName: name,
+                appIcon: icon,
+                appDescription: description
+            }
+        });
+    }
+
+    cancel() {
+        this.setState((state, props) => {
+            state.confirmID = null;
```

Comment:
Should the other state params be set to null as well?

Reply:
can't hurt, added in https://github.com/smartdevicelink/generic_hmi/pull/240/commits/b79ab73fcfd760a30437c25d53d1c6665ee29526